### PR TITLE
[MetricsAdvisor] Added AAD support

### LINKED ISSUE #17768: [MetricsAdvisor] Add AAD support
Goals:
- Add AAD constructors to both clients.
- Add validation for every endpoint.

### PR DESCRIPTION
**Pull request by kinelski**
Adding AAD support to Metrics Advisor. Changes:
- Added AAD constructors to both MA clients.
- Updated existing tests to support AAD as well. Every endpoint now has at least 1 AAD test (usually selected using a `useTokenCredential` bool test parameter).
- Updated recordings.
- Created `test-resources.json` to enable AAD credentials in live tests pipeline.

Fixes https://github.com/azure/azure-sdk-for-net/issues/17768.

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs [hunk 1]
```diff
@@ -19,9 +19,11 @@ public AnomalyAlertConfigurationLiveTests(bool isAsync) : base(isAsync)
         }
 
         [RecordedTest]
-        public async Task CreateAndGetAlertConfigurationWithWholeSeriesScope()
+        [TestCase(true)]
+        [TestCase(false)]
```
Comment (christothes) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs:23]: If this variation will be needed for every test, consider using the `ClientTestFixture` attribute with `additionalParameters` configured. 

[Here is an example](https://github.com/Azure/azure-sdk-for-net/blob/0f8dada4e59733da7b8fa15b530d68edf4ccfaa7/sdk/tables/Azure.Data.Tables/tests/TableServiceLiveTestsBase.cs#L14) of using it with Tables to create an additional permutation of each test to target both Cosmos and Storage endpoints.

Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs:23]: We're only adding the parameters in one test for each endpoint. We just need to make sure that AAD works, so we're not that concerned about validating every single possibility. We do the complete validation with the key credential only.

**Review by christothes — APPROVED** [review]

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs [hunk 2]
```diff
@@ -53,6 +53,37 @@ public MetricsAdvisorAdministrationClient(Uri endpoint, MetricsAdvisorKeyCredent
             _serviceRestClient = new AzureCognitiveServiceMetricsAdvisorRestAPIOpenAPIV2RestClient(_clientDiagnostics, pipeline, endpoint.AbsoluteUri);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetricsAdvisorClient"/> class.
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:57]: should it be `MetricsAdvisorAdministrationClient`?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs [hunk 3]
```diff
@@ -53,6 +53,37 @@ public MetricsAdvisorAdministrationClient(Uri endpoint, MetricsAdvisorKeyCredent
...
+        /// </summary>
+        /// <param name="endpoint">The endpoint to use for connecting to the Form Recognizer Azure Cognitive Service.</param>
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:59]: to the Metrics Advisor ... service

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs [hunk 4]
```diff
@@ -53,6 +53,37 @@ public MetricsAdvisorAdministrationClient(Uri endpoint, MetricsAdvisorKeyCredent
...
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetricsAdvisorClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">The endpoint to use for connecting to the Form Recognizer Azure Cognitive Service.</param>
+        /// <param name="credential">A credential used to authenticate to the Azure Service.</param>
+        /// <param name="options">A set of options to apply when configuring the client.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> or <paramref name="credential"/> is null.</exception>
+        public MetricsAdvisorAdministrationClient(Uri endpoint, TokenCredential credential, MetricsAdvisorClientsOptions options)
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:74]: same comments here for the docstring

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 5]
```diff
@@ -55,6 +55,37 @@ public MetricsAdvisorClient(Uri endpoint, MetricsAdvisorKeyCredential credential
             _serviceRestClient = new AzureCognitiveServiceMetricsAdvisorRestAPIOpenAPIV2RestClient(_clientDiagnostics, pipeline, endpoint.AbsoluteUri);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetricsAdvisorClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">The endpoint to use for connecting to the Form Recognizer Azure Cognitive Service.</param>
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:61]: no FR service :(

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs (hunk 2 above)
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:57]: Yep. It should.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs (hunk 3 above)
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:59]: I was pretty sure my copy-pasting wouldn't be exposed. I failed this time.

Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:59]: My next PR will update the README with AAD info, so I'll include these doc fixes there.

**Pull request merged**